common/src/main/java/com/graphicsfuzz/common/util/FragmentShaderJobToVertexShaderJob.java

```diff
@@ -195,6 +235,39 @@ public void visitVariablesDeclaration(VariablesDeclaration variablesDeclaration)
           }
         }
       }
+
+      // Replace discard with a null statement
+      @Override
+      public void visitDiscardStmt(DiscardStmt discardStmt) {
+        // Re
+        parentMap.getParent(discardStmt).replaceChild(discardStmt, new NullStmt());
+      }
+
+      // Replace fragment-only functions with parenthesis expression
+      @Override
+      public void visitFunctionCallExpr(FunctionCallExpr functionCallExpr) {
+        // For most of the functions, they take only one parameter and it's the same
```

Comment:
I think you should do super.visitFunctionCallExpr(functionCallExpr) here.

Then you don't need to use the loop below to visit each of the child arguments.

Can you add a test to make sure that a nested expression such as: dFdx(dFdy(...))) gets handled correctly?  I'm pretty sure it does, but doing child replacement with nesting can be tricky so would be good to be sure.  (BTW: I'm not sure that you're allowed to nest these two functions in GLES actually!)

might not get handled correctly.

The outer dFdx would get handled first, becoming dFdy(a, b).  Then the dFdY(a, b) would

Reply:
super call added, and I already had a test like that (gl_FragDepth = dFdx(0.01 + dFdy(0.02 + dFdxFine(0.03 + dFdyFine(0.04)))); etc). It shouldn't matter whether the inner or outer is handled first, because all it does is remove the function call itself, not even the parentheses.